src/main/scala/mesosphere/marathon/core/launchqueue/impl/TaskLauncherActor.scala

```diff
@@ -245,40 +319,10 @@ private class TaskLauncherActor(
       }
   }
 
-  def syncInstances(): Unit = {
-    instanceMap = instanceTracker.instancesBySpecSync.instancesMap(runSpecId).instanceMap
-    val readable = instanceMap.values
-      .map(i => s"${i.instanceId}:{condition: ${i.state.condition}, goal: ${i.state.goal}, version: ${i.runSpecVersion}, reservation: ${i.reservation}}")
-      .mkString(", ")
-    logger.info(s"Synced instance map to $readable")
-  }
-
   def syncInstance(instanceId: Instance.Id): Unit = {
-    instanceTracker.instancesBySpecSync.instance(instanceId) match {
-      case Some(instance) =>
-        instanceMap += instanceId -> instance
-
-        // This timeout is only between us creating provisioned instance operation based on received offer
-        // and that instance in provisioned state being persisted so this makes no sense for instances in all other states
-        if (!instance.isProvisioned) {
-          offerOperationAcceptTimeout.get(instanceId).foreach(_.cancel())
-          offerOperationAcceptTimeout -= instanceId
-        }
-        logger.info(s"Synced single $instanceId from InstanceTracker: $instance")
-
-        // Request delay for new run spec config.
-        if (!launchAllowedAt.contains(instance.runSpec.configRef)) {
-          // signal no interest in new offers until we get the back off delay.
-          // this makes sure that we see unused offers again that we rejected for the old configuration.
-          unregister()
-          rateLimiterActor ! RateLimiterActor.GetDelay(instance.runSpec.configRef)
-        }
-      case None =>
-        logger.info(s"Instance $instanceId does not exist in InstanceTracker - removing it from internal state.")
-        removeInstanceFromInternalState(instanceId)
-    }
-
-    manageOfferMatcherStatus(clock.now())
+    logger.debug(s"Synchronizing $instanceId")
+    context.become(syncing(instanceId))
```

Comment:
Hmm, what about timeouts here? InstanceTracker shouldn't ever crash, but what if it did in response to the query?

Reply:
Good point. We don't have a timeout in other cases when we load instances. We could add a schedule once and send a timeout.